ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/StatsUpdater.java

```diff
@@ -73,6 +69,9 @@ public void gatherStats(CompactionInfo ci, HiveConf conf, String userName, Strin
                 sb.append(")");
             }
             sb.append(" compute statistics");
+            if (ci.isMinorCompaction()) {
+                sb.append(" noscan");
```

Comment:
why `noscan` is used only in case of minor compaction?

Reply:
Minor compaction is expected to not compact too many files and hence in most scenarios only the number of files gets changed after minor compaction. Whereas large updates like major compaction needs to update all statistics (since it happens once in a while) to keep the metadata updated. Therefore the idea was to do a fast update of statistics on a minor compaction & do complete update in case of major compaction.

Reply:
Updated this to perform noscan only when autostats gather is enabled irrespective of whether minor/major compaction is triggered.